include/zephyr/bluetooth/gatt.h

```diff
@@ -1821,6 +1821,12 @@ struct bt_gatt_subscribe_params {
 #if defined(CONFIG_BT_EATT)
 	enum bt_att_chan_opt chan_opt;
 #endif /* CONFIG_BT_EATT */
+#if defined(CONFIG_BT_GATT_AUTO_RESUBSCRIBE)
+	/* Marks whether the subscription was sent on-air since the connection
+	 * was established.
+	 */
+	bool sent_sub;
+#endif	/* CONFIG_BT_GATT_AUTO_RESUBSCRIBE */
```

Comment:
What happened to the idea of putting it in `struct gatt_sub`? 

I'm not crazy about adding an internal field to the parameter struct, but if we do it this way:

1. `sent_sub` should be prefix with `_` to indicate an internal field
2. We should have a reference to this value only being valid if `BT_GATT_SUBSCRIBE_FLAG_NO_RESUB` is not set. 

Actually, since we already have something like `BT_GATT_SUBSCRIBE_FLAG_WRITE_PENDING` in the GATT flags, should we add this `sent_sub` as a flag there instead?

Reply:
`gatt_sub` wouldn't have worked I think, because it's per-device, not per-ccc so it breaks in some use-cases.

> Actually, since we already have something like `BT_GATT_SUBSCRIBE_FLAG_WRITE_PENDING` in the GATT flags, should we add this `sent_sub` as a flag there instead?

This sounds better, I'll try doing that yeah.

Reply:
> `gatt_sub` wouldn't have worked I think, because it's per-device, not per-ccc so it breaks in some use-cases.

Can you give one such case? Is there a use case where we only want to subscribe to a subset of the characteristics that are subscribed without `BT_GATT_SUBSCRIBE_FLAG_NO_RESUB`? 

Since `add_subscriptions` are only called once (or intended to be called once), I have trouble seeing how a per-device solution won't work. 

As I was writing the above, I just realized that `security_changed` may be called multiple times (e.g. going from Level 0 -> Level 2, and then from Level 2 to Level 3). Is that properly handled?

Reply:
I had it in mind but can't seem to remember now. But adding the flag as you propose is better anyway, because it doesn't use any more memory.

The two `security_changed` calls won't be a problem because if `add_subscriptions` is called once, then all the chars that have been re-sent over will have the flag set (and won't be sent again).